src/map/item_synthesis.cpp

```diff
@@ -0,0 +1,316 @@
+// Copyright (c) rAthena Dev Teams - Licensed under GNU GPL
+// For more information, see LICENCE in the main folder
+
+#include "item_synthesis.hpp"
+
+#include <algorithm>
+#include <memory>
+
+#include "../common/nullpo.hpp"
+#include "../common/showmsg.hpp"
+
+#include "log.hpp" // e_log_pick_type
+
+ItemSynthesisDatabase item_synthesis_db;
+
+void ItemSynthesisDatabase::clear() {
+	TypesafeYamlDatabase::clear();
+}
+
+const std::string ItemSynthesisDatabase::getDefaultLocation() {
+	return std::string(db_path) + "/item_synthesis.yml";
+}
+
+/**
+* Reads and parses an entry from the item_synthesis file.
+* @param node: YAML node containing the entry.
+* @return count of successfully parsed rows
+*/
+uint64 ItemSynthesisDatabase::parseBodyNode(const YAML::Node &node) {
+	uint32 id;
+
+	if (!this->asUInt32(node, "Id", id))
+		return 0;
+
+	std::shared_ptr<s_item_synthesis_db> entry = this->find(id);
+	bool exists = entry != nullptr;
+
+	if (!exists) {
+		if (!this->nodesExist(node, { "SourceNeeded", "SourceItem", "Reward" }))
+			return 0;
+
+		entry = std::make_shared<s_item_synthesis_db>();
+		entry->id = id;
+	}
+
+	if (this->nodeExists(node, "SourceNeeded")) {
+		if (!this->asUInt16(node, "SourceNeeded", entry->source_needed))
+			return 0;
+	}
+
+	if (this->nodeExists(node, "NeedRefine")) {
```

Comment:
will merge this to `NeedRefineMin` and `NeedRefineMax` just like item_upgrade.yml